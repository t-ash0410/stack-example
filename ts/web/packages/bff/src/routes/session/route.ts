import type { AuthNEnv } from '@bff/types'
import { Hono } from 'hono'
import { getHandler } from './get'

const sessionRoute = new Hono<AuthNEnv>().get('/', getHandler)

export { sessionRoute }
