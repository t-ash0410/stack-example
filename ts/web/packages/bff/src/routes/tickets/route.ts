import type { AuthNEnv } from '@bff/types'
import { Hono } from 'hono'
import { listHandler } from './list'

const ticketsRoute = new Hono<AuthNEnv>().get('/', listHandler)

export { ticketsRoute }
