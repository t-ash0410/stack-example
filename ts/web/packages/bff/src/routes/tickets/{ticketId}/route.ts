import { ticketDetailApp } from './app'
import { deleteHandlers } from './delete'
import { getHandlers } from './get'
import { updateHandlers } from './update'

const ticketDetailRoute = ticketDetailApp
  .get('/', ...getHandlers)
  .put('/', ...updateHandlers)
  .delete('/', ...deleteHandlers)

export { ticketDetailRoute }
