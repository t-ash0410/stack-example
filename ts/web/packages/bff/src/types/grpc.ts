import type { Client } from '@connectrpc/connect'
import type {
  AccountMgrService,
  TicketMgrService,
  TicketQuerierService,
} from '@stack-example/grpc'

type AccountMgrServiceClient = Client<typeof AccountMgrService>
type TicketMgrServiceClient = Client<typeof TicketMgrService>
type TicketQuerierServiceClient = Client<typeof TicketQuerierService>

export type {
  AccountMgrServiceClient,
  TicketMgrServiceClient,
  TicketQuerierServiceClient,
}
