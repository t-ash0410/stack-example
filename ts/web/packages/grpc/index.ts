export * from "./src/accountmgr/v1/service_pb"
export * from "./src/ticketmgr/v1/service_pb"
export * from "./src/ticketquerier/v1/service_pb"
