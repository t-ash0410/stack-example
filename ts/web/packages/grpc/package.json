{
  "name": "@stack-example/grpc",
  "module": "index.ts",
  "type": "module",
  "dependencies": {
    "@bufbuild/protobuf": "^2.2.3"
  },
  "devDependencies": {
    "@types/bun": "latest"
  },
  "peerDependencies": {
    "typescript": "^5.0.0"
  }
}
