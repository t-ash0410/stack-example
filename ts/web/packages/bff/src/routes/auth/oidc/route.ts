import type { DefaultEnv } from '@bff/types'
import { Hono } from 'hono'
import { sessionHandler } from './session'
import { slackHandler, slackValidator } from './slack'

const oidcRoute = new Hono<DefaultEnv>()
  .get('/session', sessionHandler)
  .get('/slack', slackValidator, slackHandler)

export { oidcRoute }
