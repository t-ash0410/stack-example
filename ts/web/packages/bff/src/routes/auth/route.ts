import type { DefaultEnv } from '@bff/types'
import { Hono } from 'hono'
import { oidcRoute } from './oidc'

const authRoute = new Hono<DefaultEnv>().route('/oidc', oidcRoute)

export { authRoute }
