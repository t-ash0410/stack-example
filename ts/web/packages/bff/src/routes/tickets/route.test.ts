import {
  afterEach,
  beforeEach,
  describe,
  expect,
  it,
  mock,
  spyOn,
} from 'bun:test'
import {
  activeUser,
  initHonoApp,
  mockTicketQuerierServiceClient,
} from '@bff/testutil'
import { create } from '@bufbuild/protobuf'
import { timestampFromDate } from '@bufbuild/protobuf/wkt'
import { QueryTicketsResponseSchema } from '@stack-example/grpc'
import { ticketsRoute } from './route'

describe('GET /', async () => {
  beforeEach(() => {
    spyOn(mockTicketQuerierServiceClient, 'queryTickets').mockResolvedValue(
      create(QueryTicketsResponseSchema, {
        tickets: [
          {
            ticketId: 'ticket-001',
            createdAt: timestampFromDate(new Date('2020-01-01T00:00:00.000Z')),
            updatedAt: timestampFromDate(new Date('2020-01-01T00:00:00.000Z')),
            createdBy: 'user-001',
            title: 'Some Ticket',
            description: 'Some ticket description.',
            deadline: timestampFromDate(new Date('2020-01-10T00:00:00.000Z')),
          },
          {
            ticketId: 'ticket-002',
            createdAt: timestampFromDate(new Date('2020-01-01T00:00:00.000Z')),
            updatedAt: timestampFromDate(new Date('2020-01-01T00:00:00.000Z')),
            createdBy: 'user-001',
            title: 'Ticket 2',
            description: 'Foo baa',
            deadline: timestampFromDate(new Date('2020-01-10T00:00:00.000Z')),
          },
        ],
      }),
    )
  })

  afterEach(() => {
    mock.restore()
  })

  it('returns tickets', async () => {
    const app = initHonoApp().route('', ticketsRoute)

    const res = await app.request('/', {
      method: 'GET',
    })

    expect(res.status).toBe(200)
    expect(await res.json()).toMatchSnapshot()

    expect(mockTicketQuerierServiceClient.queryTickets).toHaveBeenCalledTimes(1)
    expect(mockTicketQuerierServiceClient.queryTickets).toHaveBeenCalledWith({
      requestedBy: activeUser.userId,
    })
  })

  it('returns 500 error if the ticket list fails to be retrieved', async () => {
    spyOn(mockTicketQuerierServiceClient, 'queryTickets').mockRejectedValue(
      new Error('Some error'),
    )

    const app = initHonoApp().route('', ticketsRoute)

    const res = await app.request('/', {
      method: 'GET',
    })

    expect(res.status).toBe(500)

    expect(mockTicketQuerierServiceClient.queryTickets).toHaveBeenCalledTimes(1)
  })
})
