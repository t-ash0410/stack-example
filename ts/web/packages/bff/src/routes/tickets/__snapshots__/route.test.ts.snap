// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`GET / returns tickets 1`] = `
[
  {
    "createdAt": "2020-01-01T00:00:00.000Z",
    "deadline": "2020-01-10T00:00:00.000Z",
    "description": "Some ticket description.",
    "ticketId": "ticket-001",
    "title": "Some Ticket",
    "updatedAt": "2020-01-01T00:00:00.000Z",
  },
  {
    "createdAt": "2020-01-01T00:00:00.000Z",
    "deadline": "2020-01-10T00:00:00.000Z",
    "description": "Foo baa",
    "ticketId": "ticket-002",
    "title": "Ticket 2",
    "updatedAt": "2020-01-01T00:00:00.000Z",
  },
]
`;
